targets/csharp/source/PlayFabSDK/source/Qos/PlayFabQosApi.cs

```diff
@@ -0,0 +1,163 @@
+﻿namespace PlayFab.QoS
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using EventsModels;
+    using Internal;
+    using MultiplayerModels;
+
+    public class PlayFabQosApi
+    {
+        private const int DefaultTimeoutMs = 250;
+        private readonly Dictionary<AzureRegion, string> _dataCenterMap = new Dictionary<AzureRegion, string>();
+        private readonly Action<string> _logAction;
```

Comment:
Given the way that you're using this, I think you'd be better off removing _logAction entirely, and always returning the logged string as a field on the result instead.

Reply:
Yes, good suggestion. I have removed it. 

---
In reply to: [288687512](https://github.com/PlayFab/SDKGenerator/pull/535#discussion_r288687512) [](ancestors = 288687512)